src/storage/mvcc/txn.rs

```diff
@@ -19,26 +22,68 @@ pub struct GcInfo {
     pub is_completed: bool,
 }
 
+/// Get the GC fence for `key`. PTAL at `txn_types::Write::gc_fence`.
+pub struct GcFenceGetter<'a, S: Snapshot> {
```

Comment:
Using this getter brings extra costs. But I think both `CheckTxnStatus` and `CheckSecondaryLocks` search the commit record, so we can get the fence TS through this search.